demos/python_demos/image_inpainting_demo/inpainting_gui.py

```diff
@@ -0,0 +1,131 @@
+            if self.oldPoint[0]!=-1:
+                cv2.line(self.mask,self.oldPoint,(x,y),1,self.radius*2)
+            cv2.circle(self.mask,(x,y),self.radius,1,cv2.FILLED)
+            self.oldPoint=(x,y)
+
+            self.updateWindow()
+        else:
+            self.oldPoint=(-1,-1)
+
+
+    def run(self):
+        self.updateWindow()
+
+        if self.img is None:
+            print("Cannot load source image\n")
+            return
+
+        key= cv2.waitKey(1)
+        while key not in (27,ord('q'),ord('Q')):
+            if key in (ord(" "),ord("\r")):
+                self.isOriginalShown = False
+                self.showInfo("Processing...")
+
+                self.img[:,:,0:1][self.mask>0]=0
+                self.img[:,:,1:2][self.mask>0]=0
+                self.img[:,:,2:3][self.mask>0]=0
+                self.img=self.inpainter.process(self.img,self.mask)
+
+                self.showInfo("")
+                self.mask[:,:,:]=0
+                self.updateWindow()
+            elif key in (8,ord('c'),ord('C')): # Backspace or c
+                self.isOriginalShown = False
+                self.mask[:,:,:]=0
+                self.updateWindow()
+            elif key == ord('\t'):
+                self.isOriginalShown = not self.isOriginalShown
+                self.updateWindow()
+            elif key in (ord('h'),ord('H')):
+                if not self.isHelpShown:
+                    self.showInfo("Use mouse with LMB to paint\nBksp or C to clear\nSpace or Enter to inpaint\nTab to show original image\nEsc or Q to quit")
+                    self.isHelpShown=True
+                else:
+                    self.showInfo("")
+                    self.isHelpShown=False
+
+            key = cv2.waitKey(1)
+
+
+    def onTrackBar(self,x):
+        self.radius=x
+
+
+    def showInfo(self,text):
+        self.label=text
+        self.updateWindow()
+        cv2.waitKey(1) # This is required to actually paint window contents right away
+        self.isHelpShown=False # Any other label removes help from the screen
+
+
+    def updateWindow(self):
+        pad=10
+        margin=10
+        if self.isOriginalShown:
+            backbuffer=self.originalImg.copy()
+            sz = cv2.getTextSize("Original", cv2.FONT_HERSHEY_COMPLEX, 0.75,1)[0]
+            imgWidth = backbuffer.shape[1]
+            labelArea = backbuffer[margin:sz[1]+pad*2+margin,imgWidth-margin-(sz[0]+pad*2):imgWidth-margin]
+            labelArea//=2
+            cv2.putText(backbuffer,"Original",(imgWidth-margin-sz[0]-pad,margin+sz[1]+pad),cv2.FONT_HERSHEY_COMPLEX, 0.75,(128,255,128))
+        else:
+            backbuffer = self.img.copy()
+            backbuffer[:,:,0:1][self.mask>0]=self.maskColor[0]
+            backbuffer[:,:,1:2][self.mask>0]=self.maskColor[1]
+            backbuffer[:,:,2:3][self.mask>0]=self.maskColor[2]
+
+        if self.label is not None and self.label!="":
+            lines = self.label.split("\n")
+            count = len(lines)
```

Comment:
Is this variable really necessary? It's only used once, and the expression is trivial.

Reply:
formulas are quite complex, I'd like to keep this variable

Reply:
Well, okay.